infra/base-images/base-runner/test_all.py

```diff
@@ -156,9 +158,51 @@ def use_different_out_dir():
       os.environ['OUT'] = initial_out
 
 
+def test_all_centipede(directory, allowed_broken_targets_percentage):
+  """Do bad_build_check on all fuzz targets of centipede."""
+  sanitizer = os.getenv('SANITIZER')
+
+  # For all projects:
+  # Centipede always requires separate non-sanitized binaries as the main fuzz
+  # targets. Script bad_build_check will test if they can run with/without
+  # sanitized binaries and omit them from sanitizer checks.
+  # No need to test sanitized target binaries if this fails.
+  os.environ['SANITIZER'] = 'none'
+  unsanitized_centipede_passed = test_all(directory,
+                                          allowed_broken_targets_percentage)
+  if sanitizer == 'none' or not unsanitized_centipede_passed:
+    return unsanitized_centipede_passed
```

Comment:
Can you put this `os.environ['SANITIZER'] = 'none'...os.environ['SANITIZER'] = sanitizer` in a try...finally so we can ensure it always gets reset properly.